presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSingleStreamSpiller.java

```diff
@@ -124,8 +134,10 @@ private void writePages(PagesSerde.PagesWriter pagesWriter, Iterator<Page> pages
     {
         try {
             InputStream input = new FileInputStream(targetFileName.toFile());
+            memoryContext.setBytes(BUFFER_SIZE);
```

Comment:
Underlying method of `setBytes` - `com.facebook.presto.operator.DriverContext#reserveSystemMemory` is not thread safe and you are accessing it from multiple threads.